From: Matt Fleming <matt.fleming@intel.com>
Date: Sat, 27 Aug 2011 09:35:45 +0100
Subject: [PATCH 01/11] x86: Add missing bzImage fields to struct setup_header

commit 8af21e7e71d1ac56d9b66fb787a14fd66af7f5f7 upstream.

commit 37ba7ab5e33c ("x86, boot: make kernel_alignment adjustable; new
bzImage fields") introduced some new fields into the bzImage header
but struct setup_header was not updated accordingly. Add the missing
'pref_address' and 'init_size' fields.

Signed-off-by: Matt Fleming <matt.fleming@intel.com>
Link: http://lkml.kernel.org/r/1318848017-12301-1-git-send-email-matt@console-pimps.org
Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>
---
 arch/x86/include/asm/bootparam.h |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-3.2.46/arch/x86/include/asm/bootparam.h
===================================================================
--- linux-3.2.46.orig/arch/x86/include/asm/bootparam.h	2013-07-26 18:55:48.000000000 +0000
+++ linux-3.2.46/arch/x86/include/asm/bootparam.h	2013-07-26 19:19:42.000000000 +0000
@@ -64,6 +64,8 @@
 	__u32	payload_offset;
 	__u32	payload_length;
 	__u64	setup_data;
+	__u64	pref_address;
+	__u32	init_size;
 } __attribute__((packed));
 
 struct sys_desc_table {
